client/network/src/protocol.rs

```diff
@@ -1045,17 +440,22 @@ where
 					// announces substream.
 					match <Message<B> as DecodeAll>::decode_all(&mut &received_handshake[..]) {
```

Comment:
This relates to the handshaking issue #5685. Problem is that `NotificationStreamOpened` message carries the roles of the node and right now those roles are only exchanged in the `BlockAnnouncesHandshake` message, stored in `Protocol`, and the information is then copied to `NotificationStreamOpened` event of other protocols such as GRANDPA and transactions. What should happen is that GRANDPA and transactions should sent that information in their handshake and that way they'd learn about the remote node's roles. There's a bit of a bootstrapping problem though because currently all of the protocols send empty handshakes and rely on `BlockAnnouncesHandshake` to provide the necessary information. This will be fixed in #5685